parserator_web/static/js/index.js

```diff
@@ -1,2 +1,107 @@
 /* TODO: Flesh this out to connect the form to the API and render results
    in the #address-results div. */
+"use strict"
+
+const resultsTable = document.getElementById("address-results"),
+  errorMessage = document.getElementById("error-message"),
+  parseType = document.getElementById("parse-type"),
+  resultsTableBody = document.getElementById("address-results-body"),
+  addressInput = document.getElementById("address"),
+  submitButton = document.getElementById("submit"),
+  form = document.getElementById("parserator-form")
+
+submitButton.addEventListener("click", () => parseAddress(addressInput.value))
+
+form.addEventListener("submit", (e) => e.preventDefault())
+
+addressInput.addEventListener("keyup", (e) => {
+  if (e.key === "Enter") {
+    submitButton.click()
+  }
+})
+
+/**
+ * Queries the API for the given address and
+ * displays the result.
+ *
+ * @param {String} address
+ *
+ */
+async function parseAddress (address) {
+  if (address === "") {
+    displayError("Enter an address!")
+    return
+  }
+
+  const response = await fetch(`api/parse/?address=${address}`),
+    json = await response.json()
+
+  switch (json.address_type) {
+  case "Error":
+    displayError("Error: Could not parse due to a repeated label.")
```

Comment:
I was actually wondering about this as well. I only accounted for this error because the `usaddress` docs didn't outline any errors other than `RepeatedLabelError`. And in the case of no input, the endpoint isn't called.

However an ambiguous address could also be classified as an error, especially if the `usaddress.tag()` returns no components (e.g. when it receives "     " as an input). What do you think about that change?

Additionally, on account of the suggestion that the endpoint also return non-200 status codes, would you agree that it makes more sense to switch based on status codes?